src/coreclr/jit/optimizebools.cpp

```diff
@@ -962,23 +962,26 @@ bool OptBoolsDsc::optOptimizeRangeTests()
 //
 bool OptBoolsDsc::optOptimizeCompareChainCondBlock()
 {
-    assert((m_b1 != nullptr) && (m_b2 != nullptr) && (m_b3 == nullptr));
+    assert((m_b1 != nullptr) && (m_b2 != nullptr));
```

Comment:
As pointed by @jakobbotsch in https://github.com/dotnet/runtime/issues/96819#issuecomment-1886727977, should we still consider doing this based on the PGO information?

Reply:
"That would require us to have reliable PGO data at the point of "optimize bools" though, which we don't have yet"
- I think that's still the current state of PGO. So, not much we can do.

Reply:
I actually think our PGO is in a good shape now (certainly much better than it was back then). cc @amanasifkhalid @AndyAyersMS for thoughts on using PGO information during `optOptimizeBools`

Reply:
I think we ought to use it as a matter of principle, and identify cases where the PGO data fails us as motivation for introducing another profile repair call site somewhere midway through compilation. I don't think it's all that common for the profile data to be completely wrecked post-morph, and in the cases where it is, there is usually a bigger underlying problem we've yet to solve (i.e. profile data for OSR methods), but I don't think that should block us from leveraging PGO data at all.

Reply:
If we're going to use PGO in `optimize bools`, then it should be used in `if conversion` too, given they are fairly interlinked.

I'd like to suggest doing that in a separate PR, as I suspect it'll require some back and too in deciding what the exact cut off points are. Whereas this PR should only be causing diffs to happen in return cases.

Reply:
I view if-conversion and this optimization as orthogonal optimizations: converting short circuiting conditionals into bitwise ops can be beneficial even on platforms that do not benefit from if conversion.

With that said, I am ok with not switching to utilize PGO information in this PR.

Reply:
Added an issue here: https://github.com/dotnet/runtime/issues/113080

(including if conversion, to save opening two issues)